File: app_pojav_zh/src/main/java/com/movtery/utils/CopyDefaultFromAssets.java
package com.movtery.utils;

import android.content.Context;

import net.kdt.pojavlaunch.Tools;

import java.io.File;
import java.io.IOException;

public class CopyDefaultFromAssets {
    public static void copyFromAssets(Context context) throws IOException {
        //默认控制布局
        if (checkDirectoryEmpty(Tools.CTRLMAP_PATH)) {
            Tools.copyAssetFile(context, "default.json", Tools.CTRLMAP_PATH, false);
        }
        //默认虚拟鼠标
        if (checkDirectoryEmpty(PojavZHTools.DIR_CUSTOM_MOUSE)) {
            Tools.copyAssetFile(context, "default_mouse.png", PojavZHTools.DIR_CUSTOM_MOUSE, false);
        }
    }

    public static boolean checkDirectoryEmpty(String dir) {
        File controlDir = new File(dir);
        File[] files = controlDir.listFiles();
        if (files != null) {
            return files.length == 0;
        } else {
            return true;
        }
    }
}

File: app_pojav_zh/src/main/java/com/movtery/pojavzh/utils/CopyDefaultFromAssets.kt
package com.movtery.pojavzh.utils

import android.content.Context
import net.kdt.pojavlaunch.Tools
import java.io.File
import java.io.IOException

class CopyDefaultFromAssets {
    companion object {
        @JvmStatic
        @Throws(IOException::class)
        fun copyFromAssets(context: Context?) {
            //默认控制布局
            if (checkDirectoryEmpty(PathAndUrlManager.DIR_CTRLMAP_PATH)) {
                Tools.copyAssetFile(context, "default.json", PathAndUrlManager.DIR_CTRLMAP_PATH, false)
            }
        }

        private fun checkDirectoryEmpty(dir: String?): Boolean {
            val controlDir = dir?.let { File(it) }
            val files = controlDir?.listFiles()
            return files?.isEmpty() ?: true
        }
    }
}
